x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/inference_flyout_wrapper.tsx

```diff
@@ -101,23 +101,23 @@ export const InferenceFlyoutWrapper: React.FC<InferenceFlyoutWrapperProps> = ({
             isPreconfigured={isPreconfigured}
           />
           <EuiSpacer size="m" />
-          <EuiFlexGroup justifyContent="flexStart">
-            <EuiFlexItem grow={false}>
-              <EuiButton
-                fill
-                color="success"
-                size="m"
-                isLoading={form.isSubmitting || isLoading}
-                disabled={
-                  (!form.isValid && form.isSubmitted) || isLoading || isPreconfigured // Disable edit option for preconfigured endpoints
-                }
-                data-test-subj="inference-endpoint-submit-button"
-                onClick={handleSubmit}
-              >
-                {LABELS.SAVE}
-              </EuiButton>
-            </EuiFlexItem>
-          </EuiFlexGroup>
+          {isPreconfigured ? null : (
```

Comment:
Is there a `EuiHorizontalRule` which is added at the end of the `InferenceServiceFormFields` element above? When the Save button isn't showing the `hr` seems unnecessary:

<img width="1311" alt="Screenshot 2025-04-22 at 17 11 35" src="https://github.com/user-attachments/assets/427bac17-d335-49b6-a453-de99335a9c6e" />

Reply:
That horizontal rule is coming from the [`InferenceServiceFormFields` component](https://github.com/elastic/kibana/blob/main/x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/inference_service_form_fields.tsx#L424)

Looks like it's there because there's `ProviderSecretHiddenField` `and ProviderConfigHiddenField` below that point. 🤔  It's not for the `Save` button. Those fields showing up depend on the return of the providers config fields. For this PR I think it's okay to leave as is - but maybe we can dig further into including it only when those fields below it are actually shown.